 your communities  Find centralized, trusted content and collaborate around the technologies you use most.  Connect and share knowledge within a single location that is structured and easy to search.  I'm currently doing an algorithm that given a sequence, checks if there is a subsequence which sum equals a given value. If I have:  where 3 8 and 7 5 are the (sequence size, value) and 0 0 tells us we reached the end. In this case it would print :  My question is, why am I getting a Time Limit Exceeded when I submit it to Mooshak? Here goes the code:  Are the values in the sequences guaranteed to be positive (or at least non-negative)?  You only exit the loop when the subsequence was found (in fact, you exit the program); if no subsequence is ever found, your program busy waits forever, which will eventually cause a timeout.  Input goes from 0 to 1000, Core size 0 Kb Data of execute 32 Mb Execution output 500 Kb Stack of execute 1 Mb Program code 100 Kb Real time timeout 60 sec Compilation timeout 60 sec Execution timeout * 5 sec  @MaxLybbert - I think you found the problem. You should post a fix as the answer.  1 Answer 1  You have an infinite loop. Inside your top-level loop, replace this:  with:  (This is based on the comment by @MaxLybbert).  P.S. As a general rule (there are some rare exceptions), you should use from instead of calling . See the accepted answer in this thread for why. For your particular code, however, it doesn't make any difference.  By clicking “Accept all cookies”, you agree Stack Exchange can store cookies on your device and disclose information in accordance with our Cookie Policy.  